### PR TITLE
Remove OperationType from OperationHandle for simplification

### PR DESCRIPTION
**Pull request by yaooqinn**
### _Why are the changes needed?_

The Operation Type is only used for converting `TOperationType` through specific Thrift Operation APIs. So we can TOperationType in these APIs directly, such as, during calling TFrontendService.ExecuteStatement, we can set TOperationType.EXECUTE_STATEMENT at the end. There is no need to add it to OperationType.

This PR removes `OperationType` completely to simply `OperationHandle` which benefits other `kyuubi.frontend.protocols`.

The thrift protocol shall be able to be removed in following PRs.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @turboFei @ulysses-you

**Review by cxzl25 — APPROVED** [review]
Checked the PR locally. LGTM

**Review by turboFei — APPROVED** [review]
LGTM

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala [hunk 1]
```diff
@@ -30,17 +30,13 @@ import org.apache.kyuubi.metrics.MetricsConstants.{OPERATION_FAIL, OPERATION_OPE
...
-abstract class KyuubiOperation(opType: OperationType, session: Session)
-  extends AbstractOperation(opType, session) {
+abstract class KyuubiOperation(session: Session)
+  extends AbstractOperation(session) {
 
-  private val opTypeName = (opType match {
-    case OperationType.UNKNOWN_OPERATION => statement
-    case _ => opType.toString
-  }).toLowerCase
+  private val opTypeName = getClass.getSimpleName
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala:39]: nit: `super.opType`

**Review by ulysses-you — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala (hunk 1 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala]: removed this and use opType directly

File: kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationHandle.scala [hunk 2]
```diff
@@ -40,35 +38,29 @@ case class OperationHandle(
...
     tOperationHandle.setOperationId(Handle.toTHandleIdentifier(identifier))
-    tOperationHandle.setOperationType(OperationType.toTOperationType(typ))
     tOperationHandle.setHasResultSet(_hasResultSet)
     tOperationHandle
   }
 
-  override def hashCode(): Int = {
-    (Objects.hashCode(identifier) + 31) * 31 + Objects.hashCode(typ)
-  }
+  override def hashCode(): Int = Objects.hashCode(identifier) + 31
```
Comment (cfmcgrady) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationHandle.scala:45]: unnecessary override?

Comment (cfmcgrady) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationHandle.scala:45]: unnecessary `+ 31`?

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/operation/OperationHandle.scala]: updated

**Comment by yaooqinn** [comment]
@cxzl25 @turboFei @ulysses-you @cfmcgrady thanks for the review. merged to master

**Pull request closed without merging**